src/private/lidar_impl.cpp

```diff
@@ -0,0 +1,191 @@
+#include "context_impl.cpp"
+#include <vector>
+#include <libm2k/utils/utils.hpp>
+#include <libm2k/utils/channel.hpp>
+#include <libm2k/utils/devicegeneric.hpp>
+#include <libm2k/utils/devicein.hpp>
+#include <libm2k/lidar.hpp>
+#include <iio.h>
+#include <iostream>
+
+using namespace std;
+using namespace libm2k::contexts;
+using namespace libm2k::utils;
+
+class Lidar::LidarImpl : public ContextImpl {
+public:
+	LidarImpl(std::string uri, iio_context* ctx, std::string name, bool sync) :
+		ContextImpl(uri, ctx, name, sync) {
+
+		pulse_dev = new DeviceGeneric(ctx, "7c700000.axi-pulse-capture");
+		afe_dev	  = new DeviceGeneric(ctx, "ad5627");
+		adc_dev	  = new DeviceIn     (ctx, "axi-ad9094-hpc");
+
+		setAllIIoAttributesToDefaultValues();
+	}
+
+	void initialize() {
+		;
+	}
+    
+	void channelEnableDisable(string channel, bool enable) {
+		if	(channel == "voltage0")
```

Comment:
maybe here a for-loop would look nicer;
something like (pseudo-code):
```
for (i = 0; i < MAX_CHANELS; i++) {
     if (channel == "voltage" + i) {
        adc_dev->enableChannel(i, enable, false);
        return;
    }
```

Reply:
Sure. The for loop looks shorter, but `"voltage" + i` looks like a hack. I will leave it like this.

Reply:
> Sure. The for loop looks shorter, but `"voltage" + i` looks like a hack. I will leave it like this.

this can be discussable [whether it's a hack or not] ; the current implementation relies on the fact that the number of channels will always be 5 (0 to 4);

1. assuming that some newer design will use fewer channels -> this may cause weird/undetermined behavior if a customer tries to access channel "voltage4", when the max channel is `voltage1`
2. assuming that some newer design will need more channels -> this code needs to add more `if () else () ` branches;

with the approach I'm proposing, you just change the number of `MAX_CHANELS` in a single place, and make sure that the code can adapt to this configuration

Reply:
Sure. Since we're discussing "what if's" in the not so distant future, a future version might use "altvoltageX' instead of "voltageX".

My implementation looks ugly as hell, the for loop is a lot cleaner but also...'smarter'. I see this as a non-issue for now.

If you insist, I can implement it either way, no problem for me. @AlexandraTrifan, @adisuciu?